firmware/application/transmitter_model.hpp

```diff
@@ -26,77 +27,68 @@
 #include <cstdint>
 #include <cstddef>
 
-#include "receiver_model.hpp"
 #include "app_settings.hpp"
+#include "max2837.hpp"
 #include "message.hpp"
+#include "receiver_model.hpp"
 #include "rf_path.hpp"
-#include "max2837.hpp"
-#include "volume.hpp"
 #include "signal.hpp"
 
 class TransmitterModel {
    public:
+    struct settings_t {
+        uint32_t baseband_bandwidth = max283x::filter::bandwidth_minimum;
+        uint32_t sampling_rate = 3'072'000;
+        uint32_t channel_bandwidth = 1;
+        /* 35 should give approx 1m transmission range. */
+        uint8_t tx_gain_db = 35;
+        bool rf_amp = false;
+    };
+
     /* The frequency to transmit on. */
     rf::Frequency target_frequency() const;
     void set_target_frequency(rf::Frequency f);
 
-    void set_antenna_bias();
-
-    bool rf_amp() const;
-    void set_rf_amp(bool enabled);
-
-    // TODO: does this make sense on TX?
-    int32_t lna() const;
-    void set_lna(int32_t v_db);
-
     uint32_t baseband_bandwidth() const;
     void set_baseband_bandwidth(uint32_t v);
 
-    // TODO: does this make sense on TX?
-    int32_t vga() const;
-    void set_vga(int32_t v_db);
-
-    int32_t tx_gain() const;
-    void set_tx_gain(int32_t v_db);
-
     // TODO: Doesn't actually affect radio.
     uint32_t channel_bandwidth() const;
     void set_channel_bandwidth(uint32_t v);
 
-    // TODO: does this make sense on TX?
     uint32_t sampling_rate() const;
     void set_sampling_rate(uint32_t v);
 
+    uint8_t tx_gain() const;
+    void set_tx_gain(uint8_t v_db);
+
+    bool rf_amp() const;
+    void set_rf_amp(bool enabled);
+
+    void set_antenna_bias();
+
     void enable();
     void disable();
 
-    /* Sets the model values without updating the radio. */
-    void set_configuration_without_update(
-        uint32_t baseband_bandwidth,
-        uint32_t sampling_rate);
+    void initialize();
```

Comment:
Just copies over a newly constructed `settings_t` so the model is pristine again.